AnnotationsAndModifiers.md

```diff
@@ -0,0 +1,51 @@
+# Annotations and Modifiers
+
+## Convention
+
+Method/variable annotations and modifiers should go on the _same_ line as the rest of the definition. i.e.
+
+## Rationale
+
+Consistency makes PRs easier to read as well as removing any potential disagreements. 
+
+## Annotations
+
+
+```swift
+// good: On the same line
+@objc func doThing() { }
+
+// bad: Not on the same line
+@objc
+func doThing() { }
+```
+
+There is an exception for when you are renaming for Objective-C. In that case, the annotation _may_ go on the preceeding line. e.g.
+
+```swift
+@objc(XYThing)
+enum Thing: Int {
+...
+}
+```
+
+The ordering for the annotations follows [that of SwiftLint](https://realm.github.io/SwiftLint/modifier_order.html). i.e. `@objc`, `@nonobjc`, `@objcMembers`, `@IBAction`, `@IBOutlet`, `@IBDesignable`, `IBInspectable`, `@NSManaged`, `@NSCopying`
+
+
+## Modifiers
+
+Similarly, modifiers should go on the same line and follow SwiftLint conventions for order. e.g.
```

Comment:
A bit of a philosophical thing here - but I would think we prefer swift-format over SwiftLint for better alignment with the swift evolution community?

But also, if a linter or formatter can enforces this, should our documentation just be our recommended configuration file?

Reply:
Our experience shows that both are required for different things, so I wouldn't say one or the other. 

Regarding the configuration file, that's something I've been thinking about too. I'd say it's a good idea. However, to align best with the community at large, we should be sticking with defaults as far as possible. We'd likely only end up sharing the "opt-in" rules that we recommend.